lib/smooth/core/network/Wifi.cpp

```diff
@@ -39,6 +39,7 @@ namespace smooth::core::network
 {
     Wifi::Wifi()
     {
+        ip.addr = 0;
```

Comment:
do `ip{};` in the declaration in the header file instead to zero-initialize the entire struct.

Reply:
the struct `ip_addr_t` defined in "ip4_addr.h" contains only the ip address as a `u32_t` - nothing else. So, there are no other members to initialize.

Reply:
You're technically right and even though it is unlikely to change, doing ip{} ensures that if it ever changes the entire struct is still zero-initialized.

Reply:
o.k.

Reply:
if I define `get_local_mac_address` as static I also have to define `ip` static in order to access it. If `ip` is static the compiler complains:

> error: 'constexpr' needed for in-class initialization of static data member 'ip4_addr_t smooth::core::network::Wifi::ip' of non-integral type [-fpermissive]

when I want to initialize it with ip{}. So, I let the initialization in the Wifi ctor for the moment unless you have another suggestion.

Reply:
Is the call to the esp_ function costly? If not then ìp` can be put into the function that uses it and just return it by value.